nimare/meta/cbma/base.py

```diff
@@ -206,8 +207,10 @@ def _collect_ma_maps(self, coords_key="coordinates", maps_key="ma_maps", return_
 
         Returns
         -------
-        ma_maps : :obj:`numpy.ndarray`
-            2D numpy array of shape (n_studies, n_voxels) with MA values.
+        ma_maps : :obj:`numpy.ndarray`, or :obj:`sparse._coo.core.COO`
+            Return a 2D numpy array of shape (n_studies, n_voxels) with MA values,
+            if pre-generated MA maps exist. Otherwise, a 4D sparse array of shape
+            (n_studies, mask.shape) with MA maps is returned.
         """
         if maps_key in self.inputs_.keys():
             LGR.debug(f"Loading pre-generated MA maps ({maps_key}).")
```

Comment:
I think if this is a simple voxelwise niftimasker (not a labels masker or some sort of atlas masker), this should still return a sparse matrix, not a dense one. Inconsistent behavior to re-run the `fit` method twice and have the data types be different for each run.

Reply:
I'm looking into it, we can potentially extend/catch the niftimasker class to allow for a custom transformation, here is some draft code for the masking procedure (don't know if it currently prevents entire array from being loaded at once)
```
nonzero_idx = imgs.dataobj.nonzero()
all_data = imgs.dataobj[nonzero_idx]
all_coords = np.vstack(nonzero_idx)
sparse_data = sparse.COO(all_coords, all_data, shape=imgs.shape)
masked_sparse_data = sparse_data[mask_img_.dataobj.nonzero()].T
```

Reply:
I ended up using the same approach we have in `compute_ale_ma` and in `compute_kda_ma`. Now loading the pre-generated MA maps is faster and memory efficient. 
Thanks!